### PR TITLE
Diagonal matrix

### PR DESCRIPTION
**Pull request by cimentadaj**
I fixed `as_matrix` and set `diagonal` as optional (also in docs), added tests to check that it inherits the correct diagonal and replaced all of the `as_matrix` calls within other functions to `as_matrix(diagonal = 1)`  to avoid breaking code. From my side `check()` is successful. Let me know if this is ok, or do you need other changes.

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Package: corrr
 Type: Package
-Version: 0.2.1.9000
+Version: 0.3.0.9999
```
Comment (drsimonj) [inline DESCRIPTION]: Please change this back to `0.2.1.9000`. I tend to work as follows:

- Start with dev version (ending .9000) and remain on it until a CRAN release
- For CRAN release change to next version (e.g., `0.3.0`) 
- Once on CRAN, flag release on GitHub and change to dev version on next change (by appending `.9000`)

I tend to follow advice [here](http://r-pkgs.had.co.nz/release.html#release-version).

Which reminded me, the next version should actually be `1.0.0` (due to backward compatible issue). Regardless, we'll wait until ready for a CRAN release.

File: NEWS.md [hunk 2]
```diff
@@ -1,59 +1,63 @@
...
-
-## Other
-
-- `na_omit` argument in `stretch()` changed to `na.rm` to match `gather_()`.
-- Bug fixes.
-- Improvements.
-
-# corrr 0.1.0
-
+# corrr 0.3.0.9999
```
Comment (drsimonj) [inline NEWS.md]: Will also need to change this back

**Comment by drsimonj** [comment]
Aside, I've also just merged another pull request that may create some conflicts. Please pull origin and check against these.

**Comment by cimentadaj** [comment]
Ok, merged changes and switched versioning to previous version.

**Comment by drsimonj** [comment]
Thanks for fixing and another nice addition to corrr!

**Pull request merged**